src/gl/canvas2dRe.re

```diff
@@ -88,20 +99,48 @@ external lineJoin : t => LineJoin.t => unit = "" [@@bs.set];
 external miterLimit : t => float => unit = "" [@@bs.set];
 
 /* Colors, Styles, and Shadows */
-external strokeStyle : t => string => unit = "" [@@bs.set];
-external fillStyle : t => string => unit = "" [@@bs.set];
+external setFillStyle : t => 'a => unit = "" [@@bs.set];
+external setStrokeStyle: t => 'a => unit = "" [@@bs.set];
+
+let setStrokeStyle (type a) (ctx: t) (s: style a) => switch (s) {
+  | (Gradient v) => setStrokeStyle ctx v
+  | (String v) => setStrokeStyle ctx v
+  | (Pattern v) => setStrokeStyle ctx v
+};
+
+let setFillStyle (type a) (ctx: t) (s: style a) => switch (s) {
+  | (Gradient v) => setFillStyle ctx v
+  | (String v) => setFillStyle ctx v
+  | (Pattern v) => setFillStyle ctx v
+};
+
+let reifyStyle x => {
+  if (Js.typeof x == "string") {
+    String (Obj.magic x)
+  } else {
+    (Obj.magic (Gradient (Obj.magic x)))
```

Comment:
What's the first `Obj.magic` for?

You should also annotate the types whenever you use `Obj.magic` to keep the type inference form going off on its own.